orcid-scheduler-web/src/main/java/org/orcid/scheduler/loader/cli/EmailDomainToRorLoader.java

```diff
@@ -0,0 +1,187 @@
+                    dtrm.addIdWithNoParent(rorId);
+                }
+            }
+        }
+    }
+    
+    private void storeDomainToRorMap() {
+        for(DomainToRorMap element : map.values()) {
+            LOG.debug("Processing domain {}", element.getDomain());
+            // If the domain has only one entry with no parent, store that one
+            if(element.getIdsWithNoParent().size() == 1) {
+                STATUS s = emailDomainManager.createOrUpdateEmailDomain(element.getDomain(), element.getIdsWithNoParent().get(0));
+                if(STATUS.CREATED.equals(s)) {
+                    createdEntries++;
+                } else if (STATUS.UPDATED.equals(s)) {
+                    updatedEntries++;
+                }
+            } else if(element.getIdsWithParent().size() == 1) {
+                // Else, if the domain has only one entry with parent, store that one
+                STATUS s = emailDomainManager.createOrUpdateEmailDomain(element.getDomain(), element.getIdsWithParent().get(0));
+                if(STATUS.CREATED.equals(s)) {
+                    createdEntries++;
+                } else if (STATUS.UPDATED.equals(s)) {
+                    updatedEntries++;
+                }
+            } else {            
+                // Else log a warning because there is no way to provide a suggestion
+                invalidDomains.add(element.getDomain());
+            }
+        }
+        
+        if(!invalidDomains.isEmpty()) {
+            LOG.warn("The following domains couldn't be mapped ({} In total):", invalidDomains.size());
+            for(String invalidDomain : invalidDomains) {
+                LOG.warn("{}", invalidDomain);
+            }
+        }
+        LOG.info("Created entries: {}, updated entries: {}", createdEntries, updatedEntries);
+    }
+    
+    private class DomainToRorMap {
+        private String domain;
+        private List<String> idsWithParent = new ArrayList<String>();
+        private List<String> idsWithNoParent = new ArrayList<String>();
+        
+        public void setDomain(String domain) {
+            this.domain = domain;
+        }
+        
+        public String getDomain() {
+            return this.domain;
+        }
+        
+        public void addIdWithParent(String rorId) {
+            LOG.debug("Domain {} adding {} with parent flag", this.domain, rorId);
+            idsWithParent.add(rorId);
+        }
+        
+        public List<String> getIdsWithParent() {
+            return this.idsWithParent;
+        }
+        
+        public void addIdWithNoParent(String rorId) {
+            LOG.debug("Domain {} adding {} with NO parent flag", this.domain, rorId);
+            idsWithNoParent.add(rorId);
+        }
+        
+        public List<String> getIdsWithNoParent() {
+            return this.idsWithNoParent;
+        }
+    }
+    
+    public static void main(String[] args) throws IOException {
+        String filePath = args[0]; 
```

Comment:
Should we check for empty args to avoid out of bounds exception? 

 if (args.length == 0) {
            LOG.error("No arguments - Please specify the file path.");
            System.exit(1);
}